### PR TITLE
QF-3832 Add filter that shows all subscription that expire within 1, 2, or 4 weeks

### PR DESCRIPTION
**Pull request by dddpt**
I understood the expiration date field to be the `active_until` field, correct me if I am mistaken (other related fields: `current_period_until`, `requested_cancel_at`).

Screenshot from 2024-02-14 18-05-22

### CODE REVIEW

**Comment by duke-nyuki** [comment]
Task linked: [QF-3832 Add filter that shows all subscription that expire within 1, 2, or 4 weeks](https://app.clickup.com/t/2192114/QF-3832)

**Comment by suricactus** [comment]
Correct assumption, thanks!

File: docker-app/qfieldcloud/subscription/admin.py [hunk 1]
```diff
@@ -48,6 +51,24 @@ def queryset(self, request, queryset):
...
+            ("1", "Next week"),
+            ("2", "Next 2 weeks"),
+            ("4", "Next 4 weeks"),
+        ]
+
+    def queryset(self, request, queryset):
+        if self.value():
+            now = timezone.now()
+            future_date = now + timedelta(weeks=int(self.value()))
+            return queryset.filter(active_until__range=(now, future_date))
```
Comment (suricactus) [inline docker-app/qfieldcloud/subscription/admin.py:69]: Quite smart. I would have done `active_until__gt=now(), active_since__lt=future_date`, but yours is more elegant, thanks!

File: docker-app/qfieldcloud/subscription/admin.py [hunk 2]
```diff
@@ -48,6 +51,24 @@ def queryset(self, request, queryset):
...
 
+class ActiveUntilFilter(admin.SimpleListFilter):
+    title = "expiration date"
+    parameter_name = "active_until"
+
+    def lookups(self, request, model_admin):
+        return [
+            ("1", "Next week"),
+            ("2", "Next 2 weeks"),
+            ("4", "Next 4 weeks"),
```
Comment (suricactus) [inline docker-app/qfieldcloud/subscription/admin.py:62]:
```suggestion
            ("1", _("Next week")),
            ("2", _("Next 2 weeks")),
            ("4", _("Next 4 weeks")),
```

**Review by suricactus — APPROVED** [review]
Just don't forget the translations.

File: docker-app/qfieldcloud/subscription/admin.py [hunk 3]
```diff
@@ -48,6 +51,24 @@ def queryset(self, request, queryset):
         return queryset
 
 
+class ActiveUntilFilter(admin.SimpleListFilter):
+    title = "expiration date"
```
Comment (dddpt) [inline docker-app/qfieldcloud/subscription/admin.py:55]:
```suggestion
    title = _("expiration date")
```
translation

**Comment by dddpt** [comment]
> Just don't forget the translations.

(Added your suggestion to the batch and added the `_(expiration_date)` one but somehow the ones you suggested got out of the batch and I merged.😡 Did a hacky commit `7af4980` on master to finish it.)

**Pull request merged**